src/nexuscli/nexus_client.py

```diff
@@ -483,3 +498,163 @@ def upload(self, source, destination):
+            # if files are given, rename the source to match destination
+            local_relative_dir = os.path.dirname(local_relative)
+            dst_file_name = os.path.basename(local_dst)
+            local_dst = os.path.dirname(local_dst)
+            if flatten:
+                local_relative = dst_file_name
+            else:
+                local_relative = os.path.join(
+                    local_relative_dir, dst_file_name)
+
+        destination_path = py.path.local(local_dst)
+        local_absolute_path = destination_path.join(local_relative)
+
+        if create:
+            local_absolute_path.ensure(dir=remote_isdir)
+        return str(local_absolute_path)
+
+    def _local_hash_matches_remote(
+            self, file_path, remote_hash, hash_name='sha1'):
+        """
+        True if the hash for file_path matches remote_hash for the given
+        algorithm
+        """
+        local_hash = nexus_util.calculate_hash(hash_name, file_path)
+        return local_hash == remote_hash
+
+    def _should_skip_download(
+            self, download_url, download_path, artefact, nocache):
+        """False when nocache is set or local file is out-of-date"""
+        if nocache:
+            try:
+                LOG.debug('Removing {} because nocache is set\n'.format(
+                    download_path))
+                os.remove(download_path)
+            except FileNotFoundError:
+                pass
+            return False
+
+        for hash_name in ['sha1', 'md5']:
+            h = artefact.get('checksum', {}).get(hash_name)
+            if h is None:
+                continue
+
+            if self._local_hash_matches_remote(download_path, h, hash_name):
+                LOG.debug('Skipping {download_url} because local copy '
+                          '{download_path} is up-to-date\n'.format(**locals()))
+                return True
+
+        return False
+
+    def download_file(self, download_url, destination):
+        """Download an asset from Nexus artefact repository to local
+        file system.
+
+        :param download_url: fully-qualified URL to asset being downloaded.
+        :param destination: file or directory location to save downloaded
+            asset. Must be an existing directory; any exiting file in this
+            location will be overwritten.
+        :return:
+        """
+        response = self._get(download_url)
+
+        if response.status_code != 200:
+            sys.stderr.write(response.__dict__)
+            raise DownloadError(
+                'Downloading from {download_url}. '
+                'Reason: {response.reason}'.format(**locals()))
+
+        with open(destination, 'wb') as fd:
+            LOG.debug('Writing {download_url} to {destination}\n'.format(
+                **locals()))
+            for chunk in response.iter_content():
+                fd.write(chunk)
+
+    def download(self, source, destination, log_lvl='WARNING',
```

Comment:
Could you please create an issue in github to refactor this method signature? `log_lvl` doesn't belong here.

Reply:
This should not be required as now I have configured logging at application level and have refactored this method.